alg/contour.cpp

```diff
@@ -749,6 +749,32 @@ CPLErr GDALContourGenerateEx(GDALRasterBandH hBand, void *hLayer,
                         }
                     }
                 }
+
+                // Largest requested level (levels are sorted)
+                const double maxLevel{fixedLevels.back()};
+
+                // If the maximum raster value is smaller than the last requested
+                // level, select the requested level that is just above the
+                // maximum raster value
+                if (maxLevel > dfMaximum)
+                {
+                    for (size_t i = fixedLevels.size() - 1; i > 0; --i)
+                    {
+                        if (fixedLevels[i] <= dfMaximum)
+                        {
+                            dfMaximum = fixedLevels[i + 1];
+                            break;
+                        }
+                    }
+                }
+
+                // If the maximum raster value is equal to the last requested
+                // level, add a small value to the maximum to avoid skipping the
+                // last level polygons
+                if (maxLevel == dfMaximum)
+                {
+                    dfMaximum += 1e-10;
```

Comment:
wouldn't the following be more robust, especially when dealing with values with large magnitude ?
```suggestion
                    dfMaximum = std::nextafter(dfMaximum, std::numeric_limits<double>::infinity());
```